crates/bevy_render/src/mesh/mesh/conversions.rs

```diff
@@ -25,9 +25,21 @@
 //! ```
 
 use crate::mesh::VertexAttributeValues;
+use bevy_math::{IVec2, IVec3, IVec4, UVec2, UVec3, UVec4, Vec2, Vec3, Vec4};
 use bevy_utils::EnumVariantMeta;
 use thiserror::Error;
 
+macro_rules! convert_vec_of_vec {
+    ($x:expr, $t:ty, $s:literal) => {
+        unsafe {
+            // SAFETY: x is only used in this file. It is called only with x being a value of type `Vec<Vec3>` or something similar.
+            // Vec3 is a repr(C) struct containing 3 f32s, hence it is transmutable to a [f32; 3].
+            let mut v = std::mem::ManuallyDrop::new($x);
+            Vec::from_raw_parts(v.as_mut_ptr() as *mut [$t; $s], v.len(), v.capacity())
```

Comment:
Why does this not use https://docs.rs/bytemuck/latest/bytemuck/allocation/fn.cast_vec.html?

Reply:
Mainly because I wasn't aware of it at the time. I will have a look at it tomorrow.